perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -1378,4 +1387,73 @@ public void deleteSponsorLinks(PerunSession sess, User sponsor) throws InternalE
 		}
 
 	}
+
+	@Override
+	public List<Resource> getAssignedResources(PerunSession sess, User user) throws InternalErrorException {
+		try  {
+			return jdbc.query("select distinct " + resourceMappingSelectQuery + " from resources join groups_resources on resources.id=groups_resources.resource_id " +
+					" join groups on groups_resources.group_id=groups.id" +
+					" join groups_members on groups.id=groups_members.group_id " +
+					" join members on groups_members.member_id=members.id " +
+					" where members.user_id=?", RESOURCE_MAPPER, user.getId());
+		} catch (EmptyResultDataAccessException e) {
```

Comment:
Peter B. said in different commit  https://github.com/CESNET/perun/pull/2702 that this exception is not throws for jdbc.query and you said he is right. So there is probably the same problem in every method with jdbc.query.

Reply:
I have big cleanup planned for the whole impl layer, but you are right. I removed this catch branch from the newly added methods, including throws declaration for the InternalErrorException, which is runtime for some time.